addons/gamification/tests/test_challenge.py

```diff
@@ -184,6 +184,40 @@ def test_30_create_challenge_with_sum_goal(self):
             "Challenge failed to start",
         )
 
+    def test_send_report_in_ranking(self):
+        model = self.env['ir.model'].search([('model', '=', 'gamification.badge')])[0]
+        field = self.env['ir.model.fields'].search([('model', '=', 'gamification.badge'), ('name', '=', 'rule_max_number')])[0]
+
+        sum_goal = self.env['gamification.goal.definition'].create({
+            'name': 'test1',
+            'computation_mode': 'sum',
+            'model_id': model.id,
+            'field_id': field.id
+        })
+
+        challenge = self.env['gamification.challenge'].create({
+            'name': 'test1',
+            'state': 'draft',
+            'user_domain': '[("active", "=", True)]',
+            'reward_id': 1,
+            'visibility_mode': 'ranking'
+        })
+
+        self.env['gamification.challenge.line'].create({
+            'challenge_id': challenge.id,
+            'definition_id': sum_goal.id,
+            'condition': 'higher',
+            'target_goal': 1
+        })
+
+        challenge.action_start()
+        challenge.action_report_progress()
+
+        self.assertEqual(
```

Comment:
Would probably be better to freeze_date and assert the value of "last_report_date", to make sure the report was properly sent.